pallets/ddc-payouts/src/lib.rs

```diff
@@ -45,29 +86,36 @@ pub mod pallet {
 		type RuntimeEvent: From<Event<Self>> + IsType<<Self as frame_system::Config>::RuntimeEvent>;
 		#[pallet::constant]
 		type PalletId: Get<PalletId>;
+
+		type Currency: LockableCurrency<Self::AccountId, Moment = Self::BlockNumber>;
 	}
 
 	#[pallet::event]
 	#[pallet::generate_deposit(pub(crate) fn deposit_event)]
 	pub enum Event<T: Config> {
 		BillingReportInitialized { cluster_id: ClusterId, era: DdcEra },
 		ChargingStarted { cluster_id: ClusterId, era: DdcEra },
+		Charged { cluster_id: ClusterId, era: DdcEra, customer_id: T::AccountId, amount: u128 },
 		ChargingFinished { cluster_id: ClusterId, era: DdcEra },
 		RewardingStarted { cluster_id: ClusterId, era: DdcEra },
+		Rewarded { cluster_id: ClusterId, era: DdcEra, node_id: T::AccountId, amount: u128 },
 		RewardingFinished { cluster_id: ClusterId, era: DdcEra },
```

Comment:
I think in the `Rewarded` event the field `node_id` should be renamed to `node_provider_id` or `node_stash_id`, as the reward is being received by the node operator. 
As for node identifier (node_id), we use `NodePublicKey` type, which may be different from the one supported by `T::AccountId` and depends on DDC node's internal implementation